Sources/Fluent/Query/Join.swift

```diff
@@ -12,16 +12,41 @@
 public struct Join {
     /// Entity that will be accepting
     /// the joined data
-    public let local: Entity.Type
+    public let base: Entity.Type
 
     /// Entity that will be joining
-    /// the local data
-    public let foreign: Entity.Type
+    /// the base data
+    public let joined: Entity.Type
+
+    /// See Child enum
+    public let child: Child
+
+    /// Indicates which entity contains
+    /// a foreign id pointer to the other entity
+    public enum Child {
+        /// The base entity contains
+        /// a foreign id pointer to the joined data
+        ///
+        /// base        | joined
+        /// ------------+-------
+        /// joined_id   | id
+        case base
+        /// The joined entity contains
+        /// a foreign id pointer to the base data
+        /// 
+        /// base | joined
+        /// -----+--------
+        /// id   | base_id
+        ///
+        /// This is the default case.
+        case joined
+    }
 
     /// Create a new Join
-    public init(local: Entity.Type, foreign: Entity.Type) {
-        self.local = local
-        self.foreign = foreign
+    public init(base: Entity.Type, joined: Entity.Type, child: Child = .joined) {
+        self.base = base
+        self.joined = joined
+        self.child = child
```

Comment:
Do we need this, or could we always say "`base` and `joined` are connected by a `joined_id` in the `base` table and require the user to order them properly? Not sure if that'd have other implications, but it seems easier to reason about long term if there's not really a difference